Add github actions CI

hershen opened the pull request:
Add CI functionality via github actions.
Perform 3 steps:
1) Configure project with CMAKE.
2) Build project with CMAKE.
3) Run tests with ctest.

hershen committed:
Add github actions CI

Add CI functionality via github actions.
Perform 3 steps:
1) Configure project with CMAKE.
2) Build project with CMAKE.
3) Run tests with ctest.

oorijsonar commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,14 @@
+name: hersh-CI
+on: [push, pull_request]
```
Will this run when a commit is pushed to an open PR?

hershen replied:
Yes.

oorijsonar commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,14 @@
+      - name: build
+        run: cmake --build build -j2
+      - name: test
+        run: ctest --test-dir build
```
I think you need `--output-on-failure` otherwise you won't see what went wrong.

hershen replied:
Fixed.

hershen committed:
Make test failures verbose in CI

Add --output-on-failure to test step of CI in order to make test
failures verbose.

oorijsonar commented:
You should monitor how many minutes this is taking up as you could get charged for using their runner I think.

hershen commented:
It looks like the spending limit is $0 if you don't specify a payment method.
